### ISSUE TITLE
Only iPhone Xs Crash with Unity Project

### ISSUE THREAD

**Issue by iHTCboy** [naver/cafe-sdk-ios#20]
The suspicion is that `KaleidoCore.framework` caused the crash。

----

If use Unity Project Add Cafe SDK（v3 ~ v4.0.3），When started, it will immediately flash back, only iPhone Xs type devices will crash, other types are ok。

```obj-c
ProductName(1731,0x103c9ef40) malloc: *** error for object 0x1072354c0: pointer being freed was not allocated
ProductName(1731,0x103c9ef40) malloc: *** set a breakpoint in malloc_error_break to debug
```

![i8vr1 5n rq_u34n6 lnkx](https://user-images.githubusercontent.com/11418363/48125406-8ac82600-e2b9-11e8-8523-516119f95fbe.png)

If you use iPhone Xs (Max) to create a new project, there will be no crash。

The suspicion is that `KaleidoCore.framework` caused the crash。
---
Hope you can see it and repair it as soon as possible. Thank you！

**Comment by iHTCboy** [comment]
We have tried reaching unity support but they keep grouping this issue with the memory issues they are having. We don't believe they are at all related. This issue only happens on iPhone XS.

If you are using Realm or any other library that spawns C++ threads in the background then this issue will likely affect you as well. The issue only happens on iPhone XS and only when using C++ constructs to spawn a thread, if you use the C pthread functions directly then the crash does not occur.

Its not an iOS 12 or iPhone XS issue since if you create an empty app on Xcode and do these same steps the crash does not occur. It only happens when unity is somehow linked.

If you open the attached samples you will see that the crash only happens with unity.

**Comment by AlanCBaek** [comment]
Fixed at the version 4.4.7.
